pkg/scheduler/queue/queue.go

```diff
@@ -506,6 +535,41 @@ func (q *RequestQueue) processForgetDisconnectedQueriers() (resharded bool) {
 	return q.queueBroker.forgetDisconnectedQueriers(time.Now())
 }
 
+func (q *RequestQueue) SubmitRequestSent(req *SchedulerRequest) {
+	if req != nil {
+		select {
+		case q.requestsSent <- req:
+		case <-q.stopCompleted:
+		}
+	}
+}
+
+func (q *RequestQueue) processRequestSent(req *SchedulerRequest) {
+	if req != nil {
+		q.querierInflightRequests[req.Key()] = req
+		q.QueryComponentUtilization.IncrementForComponentName(req.ExpectedQueryComponentName())
+	}
+}
+
+func (q *RequestQueue) SubmitRequestCompleted(req *SchedulerRequest) {
+	if req != nil {
```

Comment:
When would `req` be `nil`?

(same applies to `processRequestCompleted` below)

Reply:
as I was canceling queries,  killing pods, etc. there are instances where cancellations come in for requests that aren't in the scheduler's `schedulerInflightRequests`, so we get nil back from `cancelRequestAndRemoveFromPending` in this pattern used to handle frontend cancels:

```golang
schedulerReq := s.cancelRequestAndRemoveFromPending(requestKey, "frontend cancelled query")
// we may not have reached SubmitRequestSent for this query, but RequestQueue will handle this case
s.requestQueue.SubmitRequestCompleted(schedulerReq)
```
We could nil check before the calls but I figured it was easiest and safest to put the nil guards in all of these methods.